jena-base/src/main/java/org/apache/jena/atlas/io/IO.java

```diff
@@ -364,4 +369,31 @@ public static String uniqueFilename(String directory, String base, String ext) {
             return null ;
         }
     }
+    
+    /** Delete everything from a {@code Path} start point, including the path itself.
+     * This function works on files or directories.
+     * This function does not follow symbolic links.
+     */  
+    public static void deleteAll(Path start) {
+        // Walks down the tree and deletes directories on the way backup.
+        try { 
+            Files.walkFileTree(start, new SimpleFileVisitor<Path>() {
```

Comment:
Could this `new SimpleFileVisitor` be broken out as a `static` inner class?

Reply:
I'm not worried about it. I'm always a bit suspicious on anon inner classes because of the occasional GC problems, but I think I'm letting my suspicions run away from me.